h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -495,9 +481,13 @@
         if( !firstChunk || bits1 == null ) { // No more data available or allowed
           // If we are mid-parse of something, act like we saw a LF to end the
           // current token.
-          if(c == CHAR_LF || c == CHAR_CR) quoteCount++;
```

Comment:
The idea here was simple: If the parse hit the end of a chunk and there was no more data available, increment quotecount by one.

This is a nonsense and does not fit current logic. This is first part of the fix. The new condition below finalizes the fix.